Main.hs

```diff
@@ -4,22 +4,51 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short [] = True
+short [_] = True
+short [_, _] = True
+short _ = False
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely [] = True
+lovely [_] = True
+lovely [_, _] = True
+lovely (x:y:z:_) = z == 14
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [(a, b, c) | c <- [1..], b <- [1..c], a <- [1..b], c^2 == a^2 + b^2]
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [ name | x <- [1..], 
+    let name = if x `mod` 3 == 0 && x `mod` 5 == 0 then "FizzBuzz"
+               else if x `mod` 5 == 0 then "Buzz"
+               else if x `mod` 3 == 0 then "Fizz"
+               else show x
+    ]
```

Comment:
It's not critical here, but in general, avoid writing the same code twice. Introduce functions for divisibility checks and use them, or just do `mod 15` in the first condition.